src/org/wojcieszko/homework/homework02/Man.java

```diff
@@ -0,0 +1,18 @@
+package org.wojcieszko.homework.homework02;
+
+abstract class Man implements iPerson, iEmployee, iStudent, iInstructor {
+    public Man(Person person, Student student, Employee employee, Instructor instructor) {
+
+            this.getFirstName() = person.setSurName();
+
+        Person manPerson = new Person(person);
```

Comment:
Już częściowo skomentowałem przy notatki.md

Po pierwsze: klasa Man ma mieć swoje pola typu Person, Student, Employee, Instruktor. Czyli **przed** konstruktorem ma być takie coś mniej więcej:
```Java
private Person person;
private Student student;
...
```
No bo Man jest kompozycją tych atrybutów, zawiera je.

Po drugie:
należy zwyczajnie zapamiętać te obiekty z argumentów konstruktora we własnych polach. Czyli w konstruktorze machnąć coś takiego:
```Java
this.person = person;
this.student = student;
...
```

Po trzecie - to mają być pola klasy, a nie lokalne zmienne w konstruktorze (teraz są lokalne zmienne). Bo wykonanie programu dojdzie do klamerki `}` i zgubią się `manPerson` (widoczność jest do klamerki). Właściwie - dlaczego `manPerson` a nie `person`?

Po czwarte: `manPerson = new Person(person);` - to jest bardzo zły i błędny sposób inicjalizacji `person` (i potem `manStudent`, `manEmployee`, `manInstructor`). Tu jest użyty konstruktor kopiujący. Czyli `manPerson` to jest nowy obiekt stworzony na podobieństwo argumentu `person`. Jeśli potem obiekt z linii numer 7 metody main() (gdzieś tam wyżej) ktoś by zmodyfikował (nazwisko zmienił), to man, który jest z niego skomponowany nie ulegnie zmianie.  
Gdyby tu użyć `manPerson = person;` to: obiekt wskazywany przez `manPerson` (jako kompozyt w klasie Man) oraz obiekt person z klasy main to jest ten sam obiekt (podstawienie to jest powielenie referencji do obiektu) i każda zmiana w jednym to jest de facto też zmiana w tym drugim (bo referencje wskazują na ten sam obiekt na stercie).